src/App.js

```diff
@@ -4,34 +4,72 @@ import podcasts from './test_data_space.json'
 import { Route, Switch, useHistory } from 'react-router-dom'
 import DetailsPage from './pages/DetailsPage/DetailsPage'
 import { useState } from 'react'
+import FavoritesPage from './pages/FavoritesPage/FavoritesPage'
 
 function App() {
   const [detailedPodcast, setDetailedPodcast] = useState()
-  const { push } = useHistory()
+  const [favoritePodcasts, setFavoritePodcasts] = useState([])
+  const [isFavorite, setIsFavorite] = useState(false)
+  const { push, goBack } = useHistory()
   return (
     <Switch>
       <Route exact path="/">
         <PodcastListPage
           podcasts={podcasts}
           onClickDetails={handleClickDetails}
+          onClickFavorites={handleClickFavoritesPage}
+        />
+      </Route>
+      <Route path="/favorites">
+        <FavoritesPage
+          favoritePodcasts={favoritePodcasts}
+          onClickDetails={handleClickDetails}
+          onClickBackFavorites={handleClickBack}
         />
       </Route>
       <Route path="/details">
         <DetailsPage
           podcast={detailedPodcast}
-          onClickDetailsBack={handleClickDetailsBack}
+          onClickDetailsBack={handleClickBack}
+          onToggleFavorite={handleToggleFavorite}
+          isFavorite={isFavorite}
         />
       </Route>
     </Switch>
   )
 
   function handleClickDetails(id) {
     setDetailedPodcast(podcasts.find(podcast => podcast.id === id))
+    setIsFavorite(favoritePodcasts.some(podcast => podcast.id === id))
```

Comment:
If I understand correctly: when clicking on the podcast card to get more details, you check whether it already is in the array of `favoritePodcasts` and set the state accordingly? Then you pass it to `DetailsPage` to determine the styles? 
I am asking because it wasn't clear to me right away :)